docs/src/views/ShorthandProps.tsx

```diff
@@ -1,37 +1,263 @@
+          data is loading, as well as what should be rendered when data is fetched.
+        </p>
+        {codeExample([
+          `<AsyncData`,
+          `  getData={fetch('https://some/url')}`,
+          `  onLoading={<div>Loading..</div>} // renders when data is loading`,
+          `  onLoaded={data => <div>Loaded Data: {data}</div>} // renders once data is fetched`,
+          `/>`,
+        ])}
+
+        <p>
+          Lets suppose that we want to render 'loading' icon before data is fetched, and switch it
+          with the icon which {code('name')} will be provided by fetched {code('data')} object.
+          Also, note that we have {code('render')} arg that knows <em>how</em> to render{' '}
+          {code('Button')}'s icons.
+        </p>
+
+        <p>Putting it all together, we will address our async data fetch case:</p>
+        {codeExample([
+          `<Button`,
+          `  icon={render => <AsyncData`,
+          `    getData={fetch('https://some/url')}`,
+          `    onLoading={<Icon name='loading' />}`,
+          `    /* using 'render' arg here to render icon from loaded name */`,
+          `    onLoaded={data => render({ name: data.iconName })}`,
+          `  />}`,
+          `/>`,
+        ])}
+
+        <p>
+          Note that this example can be further simplified, as {code('render')} can be used to
+          render 'loading' icon as well:
+        </p>
+        {codeExample([
+          `<Button`,
+          `  icon={render => <AsyncData`,
+          `    getData={fetch('https://some/url')}`,
+          `    /* using 'render' arg here to render loading icon as well */`,
+          `    onLoading={render('loading')}`,
+          `    onLoaded={data => render({ name: data.iconName })}`,
+          `  />}`,
+          `/>`,
+        ])}
+
+        {subheader(`Customizing rendered tree`)}
+        <p>
+          There is another use case that {code('render')} callback arg is very useful for - this is
+          the case where custom element's tree should be rendered for the slot, but, at the same
+          time, all the evaluated styles and accessibility behaviors should be preserved for
+          rendered custom element.
+        </p>
+
+        <p>
+          As you might recall, there is a problem that might happen when React Element is provided
+          directly as shorthand value - in that case Stardust props are not propagated to rendered
+          element, which may result in broken styles and accessibility. In order to avoid that the
+          following strategy should be considered:
+        </p>
+
+        {codeExample([
+          `<Button icon={render => render(`,
+          `  /* what to render */`,
+          `  { name: 'chess rook' },`,
+          ``,
+          `  /* how to render */`,
+          `  (ComponentType, props) => (`,
+          `    <div class='my-icon-container'>`,
+          `      <i {...props} />`,
```

Comment:
Maybe we can showcase here the usage of the styles and accessibility, like spreading the accessbility.root elements on the `i` element or using the classes generated?